src/services/common/filesystem/fileCacheManager.spec.ts

```diff
@@ -0,0 +1,197 @@
+        size: 1024 * 1024 * 10,
+        name: 'hi.png',
+        ctime: Date.now(),
+        mtime: Date.now(),
+        mode: 0,
+        path: '/non_cached/hi.png',
+        originalFilepath: '/non_cached/hi.png',
+        isFile: () => true,
+        isDirectory: () => false,
+      }));
+
+      const { cached } = await sut.cacheFile('/non_cached', 'hi.png');
+      expect(mockedFs.touch).toBeCalled();
+      expect(cached).toBe(true);
+    });
+
+    it('Should remove files in the directory by the oldest mtime if not enough space', async () => {
+      // Total 90MB
+      const itemsInDir: ReadDirItem[] = [
+        {
+          name: 'file_1.png',
+          ctime: new Date('2022/12/01'),
+          mtime: new Date('2022/12/01'),
+          path: '/non_cached/file_1.png',
+          size: 1024 * 1024 * 30,
+          isFile: () => true,
+          isDirectory: () => false,
+        },
+        {
+          name: 'file_2.png',
+          ctime: new Date('2022/11/10'),
+          mtime: new Date('2022/11/10'),
+          path: '/non_cached/file_2.png',
+          size: 1024 * 1024 * 20,
+          isFile: () => true,
+          isDirectory: () => false,
+        },
+        {
+          name: 'file_3.png',
+          ctime: new Date('2022/11/25'),
+          mtime: new Date('2022/11/25'),
+          path: '/non_cached/file_3.png',
+          size: 1024 * 1024 * 40,
+          isFile: () => true,
+          isDirectory: () => false,
+        },
+      ];
+      mockedFs.readDir.mockImplementationOnce(async () => itemsInDir);
+      mockedFs.getDirSize.mockImplementationOnce(async () =>
+        itemsInDir.reduce<number>((prev, curr) => prev + curr.size, 0),
+      );
+
+      mockedFs.unlinkIfExists.mockImplementation(async () => true);
+
+      const sut = fileCacheManagerFactory({
+        // 100MB
+        maxSpaceInBytes: 1024 * 1024 * 100,
+        // 50MB
+        maxFileSizeToCacheInBytes: 1024 * 1024 * 50,
+      });
+      // Return true when smoke testing
+      mockedFs.exists.mockImplementationOnce(async () => true);
+      await sut.init();
+      mockedFs.exists.mockImplementationOnce(async () => true);
+      mockedFs.exists.mockImplementationOnce(async () => true);
+      mockedFs.exists.mockImplementationOnce(async () => true);
+      mockedFs.exists.mockImplementationOnce(async () => false);
+      mockedFs.exists.mockImplementationOnce(async () => true);
+
+      // We will need to free 40MB at least to fit this file
+      mockedFs.statRNFS.mockImplementationOnce(async () => ({
+        // 50MB
+        size: 1024 * 1024 * 50,
+        name: 'hi.png',
+        ctime: Date.now(),
+        mtime: Date.now(),
+        mode: 0,
+        path: '/non_cached/hi.png',
+        originalFilepath: '/non_cached/hi.png',
+        isFile: () => true,
+        isDirectory: () => false,
+      }));
+
+      const { cached } = await sut.cacheFile('/non_cached', 'hi.png');
+      expect(mockedFs.unlinkIfExists).toBeCalledTimes(2);
```

Comment:
Could be improved a bit if the params for each call to `unlinkIfExists` is matched against the expected params. 

That match will virtually discard all side-effects that this test could face - like unlinking not the oldest files always, which will be a false positive case -.